src/Raven.Server/Documents/Sharding/Background/ShardedDocumentsMigrator.cs

```diff
@@ -53,15 +54,32 @@ internal async Task ExecuteMoveDocumentsAsync()
                         {
                             _token.ThrowIfCancellationRequested();
 
-                            if (bucketStats.NumberOfDocuments == 0)
+                            if (bucketStats.Size == 0)
                                 continue;
-                            
+
                             bucket = bucketStats.Bucket;
-                            
+
                             // this bucket already been migrated
-                            if(configuration.BucketMigrations.ContainsKey(bucket))
+                            if (configuration.BucketMigrations.ContainsKey(bucket))
                                 continue;
 
+                            if (bucketStats.NumberOfDocuments == 0)
+                            {
+                                var foundTombstone = false;
+                                foreach (var tombstone in _database.ShardedDocumentsStorage.RetrieveTombstonesByBucketFrom(context, bucket, 0))
+                                {
+                                    DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Shiran, DevelopmentHelper.Severity.Normal, "handle DeletedTimeSeriesRange?");
+
+                                    if (tombstone.Flags.Contain(DocumentFlags.Artificial))
+                                        continue;
+
+                                    foundTombstone = true;
+                                }
+
+                                if (foundTombstone == false)
+                                    continue;
+                            }
+
```

Comment:
instead of those `foundTombstone` & `found` vars, create a method
`bool TryFindWrongBucket(... , out int bucket, out int shardNumber)`

Reply:
done